app/Concerns/HasRatings.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+
+namespace App\Concerns;
+
+use App\Events\User\Updated;
+use App\Libraries\CERT\VATSIMUserDetails;
+use App\Models\Rating;
+use App\Models\RatingPivot;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+trait HasRatings
+{
+    public function ratings(): BelongsToMany
+    {
+        return $this->belongsToMany(
+            Rating::class,
+            "user_ratings",
+            'user_id',
+            'rating_id'
+        )->using(RatingPivot::class)
+            ->withTimestamps();
+    }
+
+    public function syncRatings(int $actRatingCode, int $pilotRatingCode)
+    {
+        $ratings = collect();
+
+        // Handle ATC rating
+        if ($actRatingCode === 0) {
+            $this->banNetwork('Network ban discovered via Cert login.');
+        } elseif ($actRatingCode > 0) {
+            $this->endNetworkBanIfHas();
+            $ratings->push(Rating::atcRatingFromID($actRatingCode));
+        }
+
+        // Attempt to find non-instructor/admin rating
+        if ($actRatingCode >= 8) {
+            $previousRating = VATSIMUserDetails::getPreviousRatingsInfo($this->id);
```

Comment:
Just to check, does this get the previous non-special rating, or just the previous? Can imagine a situation where someone goes I1 -> I3 or SUP -> ADM

Reply:
Sounds good!